### PR TITLE
Integrate partials into bazaar

### PR DESCRIPTION
**Pull request by lightclient**
Okay, I got partial working in the bazaar EE! Setting each primitive element one-by-one isn't ideal, but I will work on the partials crate to improve that interface so entire composite types can be easily replaced.

### CODE REVIEW

**Comment by axic** [comment]
Nice! Would it make sense creating this as a separate example, like `sszbazaar`?

**Comment by lightclient** [comment]
Yeah that makes sense! Would you rather I do that, or come up with a slightly different example using the partials?

**Comment by axic** [comment]
I think it's fine keeping this code as is just renaming it.

An alternative could be having a feature flag, but not sure it is worth the effort.

**Comment by lightclient** [comment]
> I think it's fine keeping this code as is just renaming it.
> 
> An alternative could be having a feature flag, but not sure it is worth the effort.

I went ahead and made it its own crate! Happy to implement a feature flag instead if you decide you prefer that route. 🙂

File: scripts/bazaar_partials/chisel.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+bazaar:
+  file: "target/wasm32-unknown-unknown/release/phase2_bazaar_partials.wasm"
+  trimexports:
+    preset: "ewasm"
+  verifyexports:
+    preset: "ewasm"
+  repack:
```
Comment (axic) [inline scripts/bazaar_partials/chisel.toml:7]: Should use `snip` instead. That removes a lot more debugging stuff.

File: scripts/bazaar_partials/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+eth2_ssz = { git = "https://github.com/c-o-l-o-r/lighthouse", branch="ssz-merkle-partial" }
+eth2_ssz_derive  = { git = "https://github.com/c-o-l-o-r/lighthouse", branch="ssz-merkle-partial" }
+ssz_types = { git = "https://github.com/c-o-l-o-r/lighthouse", branch="ssz-merkle-partial" }
+typenum = "1.10.0"
+
+[dependencies.ewasm_api]
+git = "https://github.com/ewasm/ewasm-rust-api"
+rev = "1c01982"
+default-features = false
+features = ["std", "eth2", "qimalloc"]
```
Comment (axic) [inline scripts/bazaar_partials/Cargo.toml:24]: Try using `wee_alloc` instead of `qimalloc`. This code is complex enough to require a proper allocator.

**Pull request closed without merging**